src/gsy_e/models/strategy/update_frequency.py

```diff
@@ -169,6 +172,11 @@ def _calculate_number_of_available_updates_per_slot(self) -> int:
 
     def update_and_populate_price_settings(self, area: "Area") -> None:
         """Populate the price profiles for every available time slot."""
+
+        # Handling the case where future markets are disabled during a simulation.
+        if self._time_slot_duration_in_seconds <= 0:
```

Comment:
I really do not understand this. Why would the `GlobalConfig.slot_length.seconds` be changed while runtime. Can you please explain?

Reply:
Not the slot length, but the GlobalConfig.FUTURE_MARKET_DURATION_HOURS . I encountered this when running parameterised unit tests. In this case, the future markets were enabled at the time of test collection, therefore the parameterised strategies in the tests were created with future market strategy enabled. During the test run though, some tests changed the GlobalConfig.FUTURE_MARKET_DURATION_HOURS to 0, and because the parameterised strategies were already constructed, their _future_market_strategy member contained a FutureMarketStrategy class and not a FutureMarketStrategyInterface (which essentially disables the future markets). This effect is similar to having constructed the strategy classes and while the simulation is running, to change the value of GlobalConfig.FUTURE_MARKET_DURATION_HOURS to 0. In this case, the market_cycle of the FutureMarketStrategy will still be called (because it was constructed on the test collection)  but the self._time_slot_duration_in_seconds would be 0 (because the GlobalConfig.FUTURE_MARKET_DURATION_HOURS == 0 at the time of the test run) , and this would raise the exception that we perceived in the CI. 

In order to alleviate this, I fixed the unit tests teardown methods to not revert the future market duration to 0 but to the previously configured value. This part here is an extra sanity check, in case the future markets are disabled while the simulation is running, then do not process the timeslot at all.

Reply:
Thanks a lot for the explanation! The missing link was that the `FutureTemplateStrategyBidUpdater` and  `FutureTemplateStrategyOfferUpdater` overwrite the `_time_slot_duration_in_seconds` . I only looked at the implementation 10 lines above which made me wonder.